src/server.rs

```diff
@@ -0,0 +1,145 @@
+use std::error::Error;
+use std::sync::Arc;
+
+use tokio::net::UdpSocket;
+use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use tokio::sync::Mutex;
+use tokio::task::{JoinError, JoinHandle};
+
+use crate::ScuttleButt;
+
+/// Buffer size for UDP messages.
+const BUF_SIZE: usize = 4096;
```

Comment:
This size should be big enough for most messages without having a significant memory impact, but any static buffer size runs risk of getting exceeded since the transmitted data depends on `max_transmitted_key_values` and the size of the k/v sent through the UDP server.

So it might be worth looking into a more dynamically sized buffer that is capable of spanning multiple `recv_from` calls.

Reply:
We can eventually change the max kv logic to express it as a buffer len.

Reply:
```suggestion
/// Buffer size for UDP messages.
/// This value is set to avoid UDP fragmentation.
/// https://jvns.ca/blog/2017/02/07/mtu/
const BUF_SIZE: usize = 1_472;
```

Reply:
I'll try to scuttlebutt model part to include serialization and express mtu in bytes.

Reply:
Actually can we remove the fixed-size stack buffer the moment and simply use a `Vec` at least for the moment?
The max number of tuple logic should be enough most of the time, and fragment UDP packets are supposed to deal with the rest.